src/core/comm/coll.cc

```diff
@@ -241,13 +241,13 @@ int collAllgather(
 int collInit(int argc, char* argv[])
 {
   current_unique_id = 0;
-#ifdef LEGATE_USE_GASNET
+#ifdef LEGATE_USE_NETWORK
   int provided, init_flag = 0;
   CHECK_MPI(MPI_Initialized(&init_flag));
   if (!init_flag) {
     log_coll.fatal(
       "MPI has not been initialized, it should be initialized by "
-      "GASNet");
+      "the networking backend");
```

Comment:
@eddy16112 In the future we will want to also expect the case where the networking backend doesn't initialize MPI, and in that case do it ourselves.

Reply:
Can we have one variable GASNET_CONDUIT to tell which conduit is enabled for gasnet and another one LEGATE_NETWORK to tell which network backend is used? Then we can add a more comprehensive check here to check if we need to init MPI or not.

Reply:
Would it be simpler to just query MPI directly, using the `MPI_Initialized` call?

Reply:
Yes, we do use `MPI_Initialized`. What I mean is we can have an extra level of verification. For example: if MPI_Initialized == true, then GASNET_CONDUIT should be mpi or ibv, otherwise, something is wrong.

Reply:
You can check these macros, defined in `realm_defines.h`:

```
// one of these is defined, depending on the network used:
#define REALM_USE_GASNET1
/* #undef REALM_USE_GASNETEX */
/* #undef REALM_USE_MPI */

// if using gasnet, one of these will be defined:
#define GASNET_CONDUIT_MPI 1
/* #undef GASNET_CONDUIT_IBV */
/* #undef GASNET_CONDUIT_UDP */
/* #undef GASNET_CONDUIT_ARIES */
/* #undef GASNET_CONDUIT_GEMINI */
/* #undef GASNET_CONDUIT_PSM */
/* #undef GASNET_CONDUIT_UCX */
/* #undef GASNET_CONDUIT_OFI */
```